### PR TITLE
build: simplify release process

### PR DESCRIPTION
**Pull request by rabelloo**
## Purpose

Simplify release process.

## Approach

Run 'publish' action on tags instead of branches, which makes the requirement for branches unimportant.

## Testing

After merging with main, follow the new instructions in `CONTRIBUTING.md` and check `npm` for the new version.

## Risks

1. I don't think we can protect tags in GitHub
2. Still requires branches for non-latest/diverging versions, e.g.

```
* chore(release): 2.0.1 (tag: v2.0.1)
* fix: foo
* chore(release): 2.0.0 (tag: v2.0.0)
* feat!: breaking change
|  * chore(release): 1.2.1 (tag: v1.2.1, release/1.x.x)
|  * fix: foo (tag: v1.2.1)
| /
* chore(release): 1.2.0 (tag: v1.2.0)
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,28 +72,58 @@ If you need to override SemVer behavior (not recommended):
 
 The last line of the script's log will give you the command you need to execute to push the commit and tag.
 
-### Release a major version (example)
+### Release latest version (example)
```
Comment (josokinas) [inline CONTRIBUTING.md:75]: Only suggestion:

```suggestion
### Release the latest version
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -72,28 +72,58 @@ If you need to override SemVer behavior (not recommended):
...
+# checkout latest 'main'
+git checkout main
+git pull
 
-4.  GitHub Actions will publish to registry automatically.
+# use automated release script
+npm run release
 
-5.  Create a pull request to merge the major version bump back to `main`.
+# push to trigger 'publish' GitHub Action
```
Comment (josokinas) [inline CONTRIBUTING.md:93]: Maybe:

```suggestion
# push to let newly created tag trigger 'publish' GitHub Action
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -72,28 +72,58 @@ If you need to override SemVer behavior (not recommended):
...
-3.  Run `npm run release`, then follow instructions to push.
+```sh
+# ensure tree is clean - WARNING: will delete pending changes
+git reset --hard
 
-4.  GitHub Actions will publish to registry automatically.
+# get all latest changes from remote
+git fetch --all --prune
+
+# checkout latest available version tag
```
Comment (josokinas) [inline CONTRIBUTING.md:114]: It's not the latest latest though, maybe:

```suggestion
# checkout latest available version tag for v1.x.x
```

or simply

```suggestion
# checkout latest available version tag for v1
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -72,28 +72,58 @@ If you need to override SemVer behavior (not recommended):
...
+git branch release/1.x.x
+
+# cherry-pick required fixes from `main`
+git cherry-pick c6da29b
+# pick appropriate commit hashes and repeat as needed
+
+# use automated release script
+npm run release
+
+# push to trigger 'publish' GitHub Action
```
Comment (josokinas) [inline CONTRIBUTING.md:127]: Also maybe:

```suggestion
# push to let newly created tag trigger 'publish' GitHub Action
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -72,28 +72,58 @@ If you need to override SemVer behavior (not recommended):
...
+# get all latest changes from remote
+git fetch --all --prune
+
+# checkout latest available version tag
+git checkout v1.2.0
+
+# create release branch according to format
+git branch release/1.x.x
+
+# cherry-pick required fixes from `main`
```
Comment (josokinas) [inline CONTRIBUTING.md:120]: To follow previous code:

```suggestion
# cherry-pick required fixes from 'main'
```

**Review by josokinas — APPROVED** [review]
Looks good.

Just to make it clear, in the 2nd example the tag gets pushed onto release branch and will no longer belong to `main` correct?

File: CONTRIBUTING.md (hunk 1 above)
Comment (josokinas) [reply CONTRIBUTING.md:75]: BTW, removing "example" here from title as these seem to be regular instructions, as where 2nd one is an example.

**Comment by rabelloo** [comment]
> Just to make it clear, in the 2nd example the tag gets pushed onto release branch and will no longer belong to `main` correct?

Tags are their own thing, just a pointer to a commit, they aren't associated to branches at all.

**Comment by josokinas** [comment]
> Tags are their own thing, just a pointer to a commit, they aren't associated to branches at all.

Sorry, I meant commit coming alongside tag on a release branch.

File: CONTRIBUTING.md (hunk 1 above)
Comment (josokinas) [reply CONTRIBUTING.md]: Probably native speaker could answer better, but my thinking is that "latest" is only one so it's `the` 🤔

**Comment by rabelloo** [comment]
> Sorry, I meant commit coming alongside tag on a release branch.

Yea the commit, the tag and the branch should all be pushed together at once

**Comment by josokinas** [comment]
> Yea the commit, the tag and the branch should all be pushed together at once

But then these (hotfixes) won't be on `main` right?

BTW, something's stuck on pipeline 🤔

**Comment by rabelloo** [comment]
> But then these (hotfixes) won't be on `main` right?

The commits should, since we're just cherry-picking.

It should look like this before

```
* chore(release): 2.0.1 (tag: v2.0.1)
* fix: foo
* chore(release): 2.0.0 (tag: v2.0.0)
* feat!: breaking change
* chore(release): 1.2.0 (tag: v1.2.0)
```

And like this after

```
* chore(release): 2.0.1 (tag: v2.0.1)
* fix: foo
* chore(release): 2.0.0 (tag: v2.0.0)
* feat!: breaking change
|  * chore(release): 1.2.1 (tag: v1.2.1, release/1.x.x)
|  * fix: foo (tag: v1.2.1)
| /
* chore(release): 1.2.0 (tag: v1.2.0)
```

**Pull request merged**